frontend/src/modules/member/components/view/member-view-contributions.vue

```diff
@@ -1,220 +1,613 @@
+  // If the overlap percentage is greater than the threshold, the function returns true, otherwise it returns false.
+  return overlapPercentage > percentage
+}
+
+// Update `tooltipPos`
+watch(
+  () => [targetNodePos.value, tooltipOpacity.value],
+  () => {
+    if (!graph.value || !tooltip.value) return
+
+    // translate coordinates: SVG -> DOM
+    const domPoint =
+      graph.value.translateFromSvgToDomCoordinates(
+        targetNodePos.value
+      )
+
+    tooltipPos.value = {
+      left:
+        domPoint.x -
+        tooltip.value.offsetWidth / 2 +
+        // left +
+        'px',
+      top: domPoint.y - targetNodeRadius.value - 320 + 'px'
+    }
+  },
+  { deep: true }
+)
+
+watch(
+  () => [edgeCenterPos.value, edgeToolTipOpacity.value],
+  () => {
+    if (!graph.value || !edgeTooltip.value)
+      return { x: 0, y: 0 }
+    if (!targetEdgeId.value) return { x: 0, y: 0 }
+
+    // translate coordinates: SVG -> DOM
+    const domPoint =
+      graph.value.translateFromSvgToDomCoordinates(
+        edgeCenterPos.value
+      )
+    // calculates top-left position of the tooltip.
+    edgeToolTipPos.value = {
+      left:
+        domPoint.x -
+        edgeTooltip.value.offsetWidth / 2 +
+        'px',
+      top: domPoint.y - edgeMarginTop - 60 - 10 + 'px'
+    }
+  },
+  { deep: true }
+)
+
+// Remove all popups or hovers
+function turnOff() {
+  targetNodeId.value = ''
+  tooltipOpacity.value = 0 // hide
+  hoveredNode.value = null
+  hoveredEdge.value = null
+  targetEdgeId.value = ''
+  edgeToolTipOpacity.value = 0 // hide
+}
+
+const eventHandlers = {
+  'node:click': ({ node }) => {
+    if (isDisabled.value) {
+      turnOff()
+      return
+    }
+    targetNodeId.value = node
+    tooltipOpacity.value = 1 // show
+    hoveredNode.value = nodes.value[node].name
+  },
+  'view:click': () => {
+    targetNodeId.value = ''
+    tooltipOpacity.value = 0 // hide
+    hoveredNode.value = null
+    hoveredEdge.value = null
+    targetEdgeId.value = ''
+    edgeToolTipOpacity.value = 0 // hide
+  },
+  'node:pointerover': ({ node }) => {
+    if (isDisabled.value) return
+    hoveredNode.value = nodes.value[node].name
+  },
+  'node:pointerout': () => {
+    if (isDisabled.value) return
+    hoveredNode.value = null
+  },
+  'edge:click': ({ edge }) => {
+    if (isDisabled.value) {
+      turnOff()
+      return
+    }
+    hoveredEdge.value = edges.value[edge]
+    targetEdgeId.value = edge ?? ''
+    edgeToolTipOpacity.value = 1 // show
+  },
+  'edge:pointerover': ({ edge }) => {
+    if (isDisabled.value) return
+
+    hoveredEdge.value = edges.value[edge]
+  },
+  'edge:pointerout': () => {
+    if (isDisabled.value) return
+    hoveredEdge.value = null
+  },
+  'view:zoom'(zoom) {
+    if (zoom < 0.7) {
+      configs.node.label.visible = false
+    } else {
+      configs.node.label.visible = true
+    }
+    if (zoom < 1) {
+      reduceFactor.value = zoom
+    }
+  }
+}
 </script>
+
+<style lang="css" scoped>
```

Comment:
Could you update the lang to `scss`? Otherwise we will get warnings by using the `@apply`